### ISSUE TITLE
S3906: False positive when using generic type constraint

### ISSUE THREAD

**Issue by antoinebj** [SonarSource/sonar-dotnet#2423]
### Description

The S3906 rule "Event Handlers should have the correct signature" does not recognize generic types being constrained to `EventArgs` or to a subclass of it.

### Repro steps

1. Create a method or class with a generic type parameter constrained to `EventArgs`

2. Use the constrained generic type as type parameter for `EventHandler<>` when declaring an event.

Example:
```csharp
public class Foo<TEventArgs> where TEventArgs : EventArgs
{
    public event EventHandler<TEventArgs> SomethingHappened; // S3906
}
```

### Expected behavior

No violation of the rule should be detected.

### Actual behavior

A violation of the rule is detected.

### Known workarounds

Suppressing the message.

### Related information

* SonarC# Version 7.14.0.8411
* Visual Studio Version 16.0.3

**Comment by andrei-epure-sonarsource** [comment]
thanks, @antoinebj for the issue+reproducer
